cmd/lotus-shed/datastore-vlog.go

```diff
@@ -0,0 +1,342 @@
+package main
+
+import (
+	"bufio"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash"
+	"hash/crc32"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/dgraph-io/badger/v2/y"
+	block "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-base32"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+)
+
+var datastoreVlog2CarCmd = &cli.Command{
+	Name:  "vlog2car",
+	Usage: "convert badger blockstore .vlog to .car",
+	Flags: []cli.Flag{
+		&cli.PathFlag{
+			Name:     "vlog",
```

Comment:
For my own curiosity: what is a badger vlog?  I see a bunch of `.vlog` files under datastore repos.  What is the meaning of each vlog subunit?  Is it a partition of data based on the key for example?

Reply:
It's basically an append-only (afaiu) k-v write log. The `Entry`/`header` structs below tell you more exactly what is in those files